b2/sync/folder.py

```diff
@@ -70,14 +70,52 @@ def __init__(self, root):
         """
         if not isinstance(root, six.text_type):
             raise ValueError('folder path should be unicode: %s' % repr(root))
-        self.root = os.path.abspath(root)
+
+        # Using normpath here allows windows and linux paths to be handled.
+        self.root = os.path.normpath(os.path.abspath(root))
+
+    @property
+    def prefix_len(self):
+        """
+
+            Prefix length calculation:
+                https://github.com/Backblaze/B2_Command_Line_Tool/issues/334
+
+            Using os.path.normpath and os.path.abspath, there are three cases to
+                handle here:
+                  * root paths with a drive specification (windows),
+                  * root paths without a drive specification (not windows),
+                  * non root paths
+
+            To handle the first two cases, os.path.splitdrive is used to differentiate
+                between platforms.
+
+                >>> import posixpath
+                >>> import ntpath
+                >>> posix_root = '/'
+                >>> nt_root = 'C://'
+                >>> posixpath.splitdrive(posixpath.normpath(posix_root))
+                ('', '/')
+                >>> map(len, _)
+                [0, 1]
+                >>> ntpath.splitdrive(ntpath.normpath(nt_root))
+                ('C:', '\\')
+                >>> map(len, _)
+                [2, 1]
+
+        """
+
+        drive, tail = os.path.splitdrive(self.root)
+        if len(tail) == 1:
+            return len(drive) + len(tail)
+        else:
+            return len(self.root) + 1
 
     def folder_type(self):
         return 'local'
 
     def all_files(self, reporter):
-        prefix_len = len(self.root) + 1  # include trailing '/' in prefix length
```

Comment:
We've lost the comment saying that the trailing "/" is in the prefix length.  I think that would be worth saying somewhere.

It's probably also worth saying that self.root has a trailing "/" only if it's the root of the filesystem.

Reply:
I'll add it back in. There's somewhat of an explanation within the docstring for the `prefix_len` property, but it is not as explicit. 

Thinking about this a bit more, I believe there's no reason to keep passing `prefix_len` to `_walk_relative_paths` since it's a property of the class now.